pandas/io/formats/style_render.py

```diff
@@ -1345,6 +1348,165 @@ def format_index(
+
+        >>> df.style.hide([0,1]).relabel_index(["C"])  # doctest: +SKIP
+             col
+        C      c
+
+        Using a MultiIndex
+
+        >>> midx = pd.MultiIndex.from_product([[0, 1], [0, 1], [0, 1]])
+        >>> df = pd.DataFrame({"col": list(range(8))}, index=midx)
+        >>> styler = df.style  # doctest: +SKIP
+                  col
+        0  0  0     0
+              1     1
+           1  0     2
+              1     3
+        1  0  0     4
+              1     5
+           1  0     6
+              1     7
+        >>> styler.hide((midx.get_level_values(0)==0)|(midx.get_level_values(1)==0))
+        ...  # doctest: +SKIP
+        >>> styler.hide(level=[0,1])  # doctest: +SKIP
+        >>> styler.relabel_index(["binary6", "binary7"])  # doctest: +SKIP
+                  col
+        binary6     6
+        binary7     7
+
+        We can also achieve the above by indexing first and then re-labeling
+
+        >>> styler = df.loc[[(1,1,0), (1,1,1)]].style
+        >>> styler.hide(level=[0,1]).relabel_index(["binary6", "binary7"])
+        ...  # doctest: +SKIP
+                  col
+        binary6     6
+        binary7     7
+
+        Defining a formatting function which uses an enumeration counter. Also note
+        that the value of the index key is passed in the case of string labels so it
+        can also be inserted into the label, using curly brackets (or double curly
+        brackets if the string if pre-formatted),
+
+        >>> df = pd.DataFrame({"samples": np.random.rand(10)})
+        >>> styler = df.loc[np.random.randint(0,10,3)].style
+        >>> styler.relabel_index([f"sample{i+1} ({{}})" for i in range(3)])
+        ...  # doctest: +SKIP
+                         samples
+        sample1 (5)     0.315811
+        sample2 (0)     0.495941
+        sample3 (2)     0.067946
+        """
+        axis = self.data._get_axis_number(axis)
+        if axis == 0:
+            display_funcs_, obj = self._display_funcs_index, self.index
+            hidden_labels, hidden_lvls = self.hidden_rows, self.hide_index_
+        else:
+            display_funcs_, obj = self._display_funcs_columns, self.columns
+            hidden_labels, hidden_lvls = self.hidden_columns, self.hide_columns_
+        visible_len = len(obj) - len(set(hidden_labels))
+        # visible_lvls = obj.nlevels - sum(hidden_lvls)
+        if len(labels) != visible_len:
+            raise ValueError(
+                "``labels`` must be of length equal to the number of "
+                f"visible labels along ``axis`` ({visible_len})."
+            )
+
+        if level is None:
+            level = [i for i in range(obj.nlevels) if not hidden_lvls[i]]
+        levels_ = refactor_levels(level, obj)
+
+        def alias_(x, value):
+            if isinstance(value, str):
+                return value.format(x)
+            return value
+
+        for ai, i in enumerate([i for i in range(len(obj)) if i not in hidden_labels]):
```

Comment:
IMO

```
for ai, i in enumerate(range(len(obj))):
    if i not in hidden_labels:
        ...

```

is a little easier to read

Reply:
is it much less efficient? if any?

Reply:
Ah, I know why I did it, because its enumeration: `ai` is a counter whereas `i` is an index:  `ai` might be [0, 1, 2] and `i` might be [10, 13, 201]. Serves a purpose in when setting the display funcs.

Reply:
Essentially the same, maybe even a bit more efficient because a for loop is eliminated.

Here, 1 for loop is used to filter the `i`s and then another for loop is used to iterate again over the filtered `i`s.

Reply:
``` 
for ai, i in enumerate(range(len(obj))):
    if i not in hidden_labels:
        print(ai)
```

is not equal to:

``` 
for ai, i in enumerate([n for n in range(len(obj)) if n not in hidden_labels]):
    print(ai)
```
otherwise I need,

``` 
ai = -1
for i in range(len(obj)):
    if i not in hidden_labels:
        ai += 1        
        print(ai)    
```

Reply:
Ah right, thank you're right. I forgot about the enumeration variable.